aten/src/ATen/native/Histogram.cpp

```diff
@@ -0,0 +1,133 @@
+#include <ATen/ATen.h>
+#include <ATen/Dispatch.h>
+#include <ATen/NativeFunctions.h>
+
+#include <ATen/native/Histogram.h>
+
+#include <tuple>
+
+namespace at { namespace native {
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+DEFINE_DISPATCH(histogram_stub);
+
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+DEFINE_DISPATCH(histogram_linear_stub);
+
+namespace {
+
+inline void histogram_pre_check(const Tensor& input, const Tensor& bins, const c10::optional<Tensor>& weight,
```

Comment:
Add a comment for each function describing what it does and its intended use

For functions like this the comment can be very brief